pkg/cmd/server/api/types.go

```diff
@@ -474,6 +473,17 @@ type AuditConfig struct {
 	MaximumRetainedFiles int
 	// Maximum size in megabytes of the log file before it gets rotated. Defaults to 100MB.
 	MaximumFileSizeMegabytes int
+
+	// Path to the file that defines the audit policy configuration.
+	PolicyFile string
```

Comment:
Policy files don't contain any sensitive data, right?  If you had a similar config for an apiserver upstream, would you still use an external file reference?

Reply:
Upstream apiserver uses an external file, so do we. See https://github.com/kubernetes/kubernetes/blob/v1.7.0/cluster/gce/gci/configure-helper.sh#L490 as a reference. There are no sensitive data, this is a list of rules which requests should be logged at what level.

Reply:
> Upstream apiserver uses an external file, so do we. See https://github.com/kubernetes/kubernetes/blob/v1.7.0/cluster/gce/gci/configure-helper.sh#L490 as a reference. There are no sensitive data, this is a list of rules which requests should be logged at what level.

At the point where it is wired up we use an external file.  At the point where our config is described, what is the benefit to a cluster-admin of managing a separate file?

Reply:
It might get big and I think it's more handy to manage a totally separate file. Especially that this isn't the first one, based on what I've checked.

Reply:
> It might get big and I think it's more handy to manage a totally separate file. Especially that this isn't the first one, based on what I've checked.

Others contain secrets.  Of ones which don't contain secrets, I know of the scheduler file (used by a process we don't own) and some kubelet ones (also used by processes we don't own).

Reply:
@smarterclayton convinced me that this could be a file given potential future direction, but it still seems like a shame.  See this other page for a complete response: xxxx.

Reply:
In that same way we don't own audit and we won't anymore.

Reply:
> In that same way we don't own audit and we won't anymore.

That doesn't control our config file format.  If we believed that, we'd have flags controlling our certs and flags for managing admission.  The decision about how to expose the config for our server isn't based on how kube manages its config, but based upon how we'd like to manage it.

Owning the code backing audit does not mean that everyone must choose to expose it half in flags and half in files.  You can see you aren't building that here.